protocol/0075-PLUP_protocol_upgrades.md

```diff
@@ -0,0 +1,174 @@
+
+## Spam (<a name="0075-COSMICELEVATOR-022" href="#0075-COSMICELEVATOR-022">0075-COSMICELEVATOR-022</a>) 
+   - Excessive numbers of proposals from a single validator within an epoch should be detected and rejected - (Future requirement)
+   
+## Snapshots
+   - (<a name="0075-PLUP-023" href="#0075-PLUP-023">0075-PLUP-023</a>) Post a validator becoming a tendermint validator they should be immediately allowed to propose an upgrade and be included in the overall total count
+   - (<a name="0075-PLUP-024" href="#0075-PLUP-024">0075-PLUP-024</a>) Ensure that required majority is not met when enough validators join between validator proposals and target block, i.e: In a network with 5 validators, required majority is two thirds, 4 vote to upgrade, 2 more validators join before upgrade block and do not vote. Upgrade does not take place.
+   - (<a name="0075-PLUP-025" href="#0075-PLUP-025">0075-PLUP-025</a>) Node starting from snapshot which has a proposal at a given block, ensure during replay when the block height is reached a new version is loaded and also post load an upgrade takes place at target block.
+
+## LNL Checkpoints
+   - (<a name="0075-PLUP-026" href="#0075-PLUP-026">0075-PLUP-026</a>) Validator proposals should not be stored in the checkpoints and restored into the network
+   - (<a name="0075-PLUP-027" href="#0075-PLUP-027">0075-PLUP-027</a>) Upgrade will not occur after a post checkpoint restore until new proposals are made and block height reached
+   
+## API
+   - (<a name="0075-PLUP-028" href="#0075-PLUP-028">0075-PLUP-028</a>) An datanode API should be available to provide information on the upcoming confirmed proposal including total proposals/block details/versions
+   
+
+### Successful upgrade  (<a name="0075-PLUP-029" href="#0075-PLUP-029">0075-PLUP-029</a>)
+  - A new release is made available, and is successfully deployed
+  - Setup a network with 5 validators running version x
+  - Have 4 validator submit request to upgrade to release >x at block height 1000
+  - At the end of block height 1000 a snapshot is taken and vega is stopped by the vegavisor
+  - All nodes are starting from the snapshot of block 1000 and the network resumes with version >x
+ 
+ 
+### Failing consensus
+  - (<a name="0075-PLUP-030" href="#0075-PLUP-030">0075-PLUP-030</a>) Upgrade takes place at block N. Restart with a number of validators whose voting power is <= two thirds. Restart one more validator whose voting power would take the total voting power >= two thirds, with an incorrect version. Consensus is not achieved. Now restart that validator with the correct version. Consensus is achieved.
+  - (<a name="0075-PLUP-031" href="#0075-PLUP-031">0075-PLUP-031</a>) 5 validator network. Upgrade takes places at block N. Start 3 validators immediately. Allow several blocks to pass. Start two remaining validators. (All validators continue to work) - no blocks producing as 3 validators do not have enough weight - need 70% weight to produce blocks
```

Comment:
Allow several blocks to pass. ... no blocks producing as 3 validators do not have enough weight  

This is confusing. Do we mean allow *time* to pass first.

Reply:
Yes, I never saw that. I'll correct it.